tonic/src/transport/channel/service/add_origin.rs

```diff
@@ -54,8 +54,8 @@ where
             // Split the request URI into parts.
             let mut uri: http::uri::Parts = head.uri.into();
             // Update the URI parts, setting hte scheme and authority
-            uri.scheme = self.scheme.clone();
-            uri.authority = self.authority.clone();
+            uri.scheme.clone_from(&self.scheme);
+            uri.authority.clone_from(&self.authority);
```

Comment:
What I wrote setting it as `warn` means setting `assigning_clones` as `warn` explicitly in crate wide config preparing for the future consistency as it will be allowed as default and will not be detected in the future.

Reply:
What I wrote setting it as `warn` was intended to mean setting `assigning_clones` as `warn` explicitly in crate wide config preparing for the future consistency as it will be allowed as default and will not be detected in the future.

Reply:
I understood what you meant and disagreed -- I was convinced by the reasoning upstream (which applies here, too) that there's often not an actual performance benefit and it's harder to understand. So alternatively we could maybe add a local `#[allow()]` here.

Reply:
What I have said assumes that you suggest to apply this change since you are proposing it in this pull request. I am not proposing adopting this rule. What I mean is just it seems no problem to me and if we adopt this change as you are proposing here, I think that we should maintain consistency in applying the rule. From your description it seems like you are saying that this change should not be applied, but that seems to contradict what is suggested actually in the code. Could you please clarify what you are proposing?

Reply:
I've rolled back this change and added a crate-level `allow(clippy::assigning_clones)`.